utils/transformData.js

```diff
@@ -0,0 +1,87 @@
+const getDateFormat = (date) => {
+  const stringDate = `${date.getDate()}. ${
+    date.getMonth() + 1
+  }. ${date.getFullYear()}`;
+
+  return stringDate;
+};
+
+const dateOfPublicationFormat = (data) =>
+  data.map((item) => {
+    if (item.dateOfPublication) {
+      const transformedDate = getDateFormat(
+        new Date(item.dateOfPublication.toDate()),
+      );
+
+      return { ...item, dateOfPublication: transformedDate };
+    }
+
+    return item;
+  });
+
+export const transformSnapshot = (snapshot) => {
+  const transformedData = snapshot.docs.map((doc) => ({
+    ...doc.data(),
+    id: doc.id,
+  }));
+
+  return dateOfPublicationFormat(transformedData);
+};
+
+export const groupByFirstChar = (array, groupProperty) =>
+  array.reduce((acc, item) => {
+    const firstChar = item[groupProperty].charAt(0);
+
+    if (!acc[firstChar]) {
+      acc[firstChar] = []; // mutation
+    }
+
+    acc[firstChar].push(item); // even this mutates
+
+    return acc;
+  }, {});
+
+export const createSlugLink = (title) =>
+  title
+    .normalize('NFD')
+    .replace(/[\u0300-\u036f]/g, '')
+    .replace(/\s+/g, '-')
+    .toLowerCase();
```

Comment:
Maybe the first one - https://stackoverflow.com/a/1054862? 

What is `normalize('NFD')` ? :))) What this `\u0300-\u036f` hellofathing stands for ?

Reply:
I want to say that I found this more elegant, than the for-cycle like this https://stackoverflow.com/a/5782563/14100196 
But I know, that you will tell me, that there is a one simple function, that do everything above. :D I look forward to your comment.

Reply:
The for-cycle is ugly. It actually makes sense what you did. I don't know if there is anything better. This makes sense